extensions/amp-story/1.0/amp-story.js

```diff
@@ -1277,6 +1285,20 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
+    // TODO(#37285): add SubscriptionService to actually trigger the subscription dialog.
+    const subscriptionsSection = targetPage.element.getAttribute(
+      'subscriptions-section'
+    );
+    if (
+      subscriptionsSection == 'content' ||
+      subscriptionsSection == 'limited-content'
+    ) {
```

Comment:
Plz write a helper in amp-story-page, it'd be great if you could write

```
if (targetPage.isPaywallProtected()) {}
```

Reply:
Also, how do you know if the user has the right authorizations or not? This would block no matter if the user is logged in?

Reply:
Done. Yes that logic would be added in the next PR with all the paywall triggering logic. In this PR I just want to show how to propagate the events to handle it in new extension.